File: Serene/Serene.Script/Imports/ServiceContracts/Meeting.MeetingDecisionRow.cs

namespace Serene.Meeting
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;

    [Imported, Serializable, PreserveMemberCase]
    public partial class MeetingDecisionRow
    {
        public Int32? MeetingDecisionId { get; set; }
        public Int32? MeetingId { get; set; }
        public Int32? ResolutionStatusId { get; set; }
        public Int32? DisplayOrder { get; set; }
        public Int16? IsActive { get; set; }
        public Int32? MeetingAgendaId { get; set; }
        public String Description { get; set; }
        public Int32? ImageFileId { get; set; }
        public Int32? ResponsibleContactId { get; set; }
        public String DueDate { get; set; }
        public String MeetingMeetingGuid { get; set; }
        public String MeetingMeetingName { get; set; }
        public Int32? MeetingMeetingTypeId { get; set; }
        public String MeetingStartDate { get; set; }
        public String MeetingEndDate { get; set; }
        public Int32? MeetingInsertUserId { get; set; }
        public Int32? MeetingInsertDate { get; set; }
        public Int16? MeetingIsActive { get; set; }
        public Int32? MeetingUpdateUserId { get; set; }
        public Int32? MeetingUpdateDate { get; set; }
        public Int32? MeetingLocationId { get; set; }
        public Int32? MeetingOrganizationId { get; set; }
        public Int32? MeetingOrganizerContactId { get; set; }
        public Int32? MeetingReporterContactId { get; set; }
        public Int32? MeetingAgendaAgendaTypeId { get; set; }
        public Int32? MeetingAgendaMeetingId { get; set; }
        public Int32? MeetingAgendaDisplayOrder { get; set; }
        public Int16? MeetingAgendaIsActive { get; set; }
        public Int32? MeetingAgendaRequestedByContactId { get; set; }
        public String MeetingAgendaTags { get; set; }
        public String MeetingAgendaTitle { get; set; }
        public String MeetingAgendaDescription { get; set; }
        public Int32? MeetingAgendaImageFileId { get; set; }
        public String ResponsibleContactDisplayName { get; set; }
        public String ResponsibleContactEmail { get; set; }
        public Int16? ResponsibleContactIsActive { get; set; }
        public String ResponsibleContactTitle { get; set; }
        public String ResponsibleContactFirstname { get; set; }
        public String ResponsibleContactLastname { get; set; }
    
        [Imported, PreserveMemberCase]
        public static class Fields
        {
            [InlineConstant] public const string MeetingDecisionId = "MeetingDecisionId";
            [InlineConstant] public const string MeetingId = "MeetingId";
            [InlineConstant] public const string ResolutionStatusId = "ResolutionStatusId";
            [InlineConstant] public const string DisplayOrder = "DisplayOrder";
            [InlineConstant] public const string IsActive = "IsActive";
            [InlineConstant] public const string MeetingAgendaId = "MeetingAgendaId";
            [InlineConstant] public const string Description = "Description";
            [InlineConstant] public const string ImageFileId = "ImageFileId";
            [InlineConstant] public const string ResponsibleContactId = "ResponsibleContactId";
            [InlineConstant] public const string DueDate = "DueDate";
            [InlineConstant] public const string MeetingMeetingGuid = "MeetingMeetingGuid";
            [InlineConstant] public const string MeetingMeetingName = "MeetingMeetingName";
            [InlineConstant] public const string MeetingMeetingTypeId = "MeetingMeetingTypeId";
            [InlineConstant] public const string MeetingStartDate = "MeetingStartDate";
            [InlineConstant] public const string MeetingEndDate = "MeetingEndDate";
            [InlineConstant] public const string MeetingInsertUserId = "MeetingInsertUserId";
            [InlineConstant] public const string MeetingInsertDate = "MeetingInsertDate";
            [InlineConstant] public const string MeetingIsActive = "MeetingIsActive";
            [InlineConstant] public const string MeetingUpdateUserId = "MeetingUpdateUserId";
            [InlineConstant] public const string MeetingUpdateDate = "MeetingUpdateDate";
            [InlineConstant] public const string MeetingLocationId = "MeetingLocationId";
            [InlineConstant] public const string MeetingOrganizationId = "MeetingOrganizationId";
            [InlineConstant] public const string MeetingOrganizerContactId = "MeetingOrganizerContactId";
            [InlineConstant] public const string MeetingReporterContactId = "MeetingReporterContactId";
            [InlineConstant] public const string MeetingAgendaAgendaTypeId = "MeetingAgendaAgendaTypeId";
            [InlineConstant] public const string MeetingAgendaMeetingId = "MeetingAgendaMeetingId";
            [InlineConstant] public const string MeetingAgendaDisplayOrder = "MeetingAgendaDisplayOrder";
            [InlineConstant] public const string MeetingAgendaIsActive = "MeetingAgendaIsActive";
            [InlineConstant] public const string MeetingAgendaRequestedByContactId = "MeetingAgendaRequestedByContactId";
            [InlineConstant] public const string MeetingAgendaTags = "MeetingAgendaTags";
            [InlineConstant] public const string MeetingAgendaTitle = "MeetingAgendaTitle";
            [InlineConstant] public const string MeetingAgendaDescription = "MeetingAgendaDescription";
            [InlineConstant] public const string MeetingAgendaImageFileId = "MeetingAgendaImageFileId";
            [InlineConstant] public const string ResponsibleContactDisplayName = "ResponsibleContactDisplayName";
            [InlineConstant] public const string ResponsibleContactEmail = "ResponsibleContactEmail";
            [InlineConstant] public const string ResponsibleContactIsActive = "ResponsibleContactIsActive";
            [InlineConstant] public const string ResponsibleContactTitle = "ResponsibleContactTitle";
            [InlineConstant] public const string ResponsibleContactFirstname = "ResponsibleContactFirstname";
            [InlineConstant] public const string ResponsibleContactLastname = "ResponsibleContactLastname";
        }
    }
    
}

File: Serene/Serene.AspNetCore/Modules/Common/Imports/ServerTypings/Meeting.MeetingDecisionRow.ts
namespace Serene.Meeting {
    export interface MeetingDecisionRow {
        DecisionId?: number;
        MeetingId?: number;
        AgendaId?: number;
        Description?: string;
        DecisionNumber?: number;
        ResponsibleContactId?: number;
        DueDate?: string;
        ResolutionStatus?: number;
        Images?: string;
        Attachments?: string;
        MeetingMeetingName?: string;
        MeetingMeetingNumber?: string;
        MeetingMeetingGuid?: string;
        MeetingMeetingTypeId?: number;
        MeetingStartDate?: string;
        MeetingEndDate?: string;
        MeetingLocationId?: number;
        MeetingUnitId?: number;
        MeetingOrganizerContactId?: number;
        MeetingReporterContactId?: number;
        MeetingInsertUserId?: number;
        MeetingInsertDate?: string;
        MeetingUpdateUserId?: number;
        MeetingUpdateDate?: string;
        AgendaMeetingId?: number;
        AgendaNumber?: number;
        AgendaTitle?: string;
        AgendaDescription?: string;
        AgendaAgendaTypeId?: number;
        AgendaRequestedByContactId?: number;
        AgendaImages?: string;
        AgendaAttachments?: string;
        DecisionNumberName?: string;
        ResponsibleContactTitle?: string;
        ResponsibleContactFirstName?: string;
        ResponsibleContactLastName?: string;
        ResponsibleContactFullName?: string;
        ResponsibleContactEmail?: string;
        ResponsibleContactIdentityNo?: string;
        ResponsibleContactUserId?: number;
    }

    export namespace MeetingDecisionRow {
        export const idProperty = 'DecisionId';
        export const nameProperty = 'Description';
        export const localTextPrefix = 'Meeting.MeetingDecision';

        export namespace Fields {
            export declare const DecisionId: string;
            export declare const MeetingId: string;
            export declare const AgendaId: string;
            export declare const Description: string;
            export declare const DecisionNumber: string;
            export declare const ResponsibleContactId: string;
            export declare const DueDate: string;
            export declare const ResolutionStatus: string;
            export declare const Images: string;
            export declare const Attachments: string;
            export declare const MeetingMeetingName: string;
            export declare const MeetingMeetingNumber: string;
            export declare const MeetingMeetingGuid: string;
            export declare const MeetingMeetingTypeId: string;
            export declare const MeetingStartDate: string;
            export declare const MeetingEndDate: string;
            export declare const MeetingLocationId: string;
            export declare const MeetingUnitId: string;
            export declare const MeetingOrganizerContactId: string;
            export declare const MeetingReporterContactId: string;
            export declare const MeetingInsertUserId: string;
            export declare const MeetingInsertDate: string;
            export declare const MeetingUpdateUserId: string;
            export declare const MeetingUpdateDate: string;
            export declare const AgendaMeetingId: string;
            export declare const AgendaNumber: string;
            export declare const AgendaTitle: string;
            export declare const AgendaDescription: string;
            export declare const AgendaAgendaTypeId: string;
            export declare const AgendaRequestedByContactId: string;
            export declare const AgendaImages: string;
            export declare const AgendaAttachments: string;
            export declare const DecisionNumberName: string;
            export declare const ResponsibleContactTitle: string;
            export declare const ResponsibleContactFirstName: string;
            export declare const ResponsibleContactLastName: string;
            export declare const ResponsibleContactFullName: string;
            export declare const ResponsibleContactEmail: string;
            export declare const ResponsibleContactIdentityNo: string;
            export declare const ResponsibleContactUserId: string;
        }

        ['DecisionId', 'MeetingId', 'AgendaId', 'Description', 'DecisionNumber', 'ResponsibleContactId', 'DueDate', 'ResolutionStatus', 'Images', 'Attachments', 'MeetingMeetingName', 'MeetingMeetingNumber', 'MeetingMeetingGuid', 'MeetingMeetingTypeId', 'MeetingStartDate', 'MeetingEndDate', 'MeetingLocationId', 'MeetingUnitId', 'MeetingOrganizerContactId', 'MeetingReporterContactId', 'MeetingInsertUserId', 'MeetingInsertDate', 'MeetingUpdateUserId', 'MeetingUpdateDate', 'AgendaMeetingId', 'AgendaNumber', 'AgendaTitle', 'AgendaDescription', 'AgendaAgendaTypeId', 'AgendaRequestedByContactId', 'AgendaImages', 'AgendaAttachments', 'DecisionNumberName', 'ResponsibleContactTitle', 'ResponsibleContactFirstName', 'ResponsibleContactLastName', 'ResponsibleContactFullName', 'ResponsibleContactEmail', 'ResponsibleContactIdentityNo', 'ResponsibleContactUserId'].forEach(x => (<any>Fields)[x] = x);
    }
}
